hopp/simulation/technologies/tidal/mhk_tidal_plant.py

```diff
@@ -0,0 +1,235 @@
+    loss_resource_overprediction: float = field(default=0., validator=range_val(0, 100))
+    loss_transmission: float = field(default=0., validator=range_val(0, 100))
+    loss_downtime: float = field(default=0., validator=range_val(0, 100))
+    loss_additional: float = field(default=0., validator=range_val(0, 100))
+    name: str = field(default="MHKTidalPlant")
+
+
+@define
+class MHKTidalPlant(PowerSource):
+    """
+    Marine Hydrokinetic (MHK) Tidal Plant.
+
+    Args:
+        site: Site information
+        config: MHK system configuration parameters
+        cost_model_inputs: An optional dictionary containing input parameters for
+            cost modeling.
+
+        """
+    site: SiteInfo
+    config: MHKTidalConfig
+    cost_model_inputs: Optional[MHKCostModelInputs] = field(default=None)
+    config_name: str = field(default="MhkWave")
+
+    mhk_costs: Optional[MHKCosts] = field(init=False)
+
+    def __attrs_post_init__(self):
+        system_model = MhkTidal.new()
+
+        if isinstance(self.config.fin_model, dict):
+            financial_model = CustomFinancialModel(self.config.fin_model, name=self.config.name)
+        else:
+            financial_model = self.config.fin_model
+
+        financial_model = self.import_financial_model(financial_model, system_model, self.config_name)
+        
+        if self.cost_model_inputs is not None:
+            self.mhk_costs = MHKCosts(self.config, self.cost_model_inputs)
+        else:
+            self.mhk_costs = None
+            
+        super().__init__("MHKTidalPlant", self.site, system_model, financial_model)
+
+        # Set tidal resource model choice
+        system_model.MHKTidal.tidal_resource_model_choice = 1 # Time-series data=1 JPD=0
+
+        # Copy values from self.site.tidal_resource.data to system_model.MHKTidal
+        attributes_to_copy = ['tidal_velocity']
+        for attribute in attributes_to_copy:
+            setattr(system_model.MHKTidal, attribute, self.site.tidal_resource.data[attribute])
+
+        # System parameter inputs
+        self._system_model.device_rated_power = self.config.device_rating_kw
+        self._system_model.value("number_devices",  self.config.num_devices)
+        self._system_model.value("tidal_power_curve", self.config.tidal_power_curve)
+        self._system_model.value("tidal_resource", self.config.tidal_resource)
+
+        # Losses
+        loss_attributes = [
+            'loss_array_spacing', 
+            'loss_downtime', 
+            'loss_resource_overprediction', 
+            'loss_transmission', 
+            'loss_additional'
+            ]
+
+        for attribute in loss_attributes:
+            if attribute in self.config.as_dict().keys():
+                attr = getattr(self.config, attribute)
+                setattr(self._system_model.MHKTidal, attribute, attr)
+            else:
+                setattr(self._system_model.MHKTidal, attribute, 0)
```

Comment:
```suggestion
        for attribute in loss_attributes:
            attr = getattr(self.config, attribute, 0)
            setattr(self._system_model.MHKTidal, attribute, attr)
```

Reply:
You can provide a default for `getattr` to slightly simplify this: https://docs.python.org/3/library/functions.html#getattr